wallet-c/Cargo.toml

```diff
@@ -6,7 +6,10 @@ edition = "2018"
 license = "MIT OR Apache-2.0"
 
 [lib]
-crate-type = [ "staticlib", "cdylib" ]
+crate-type = ["lib"]
+
+[build]
+crate-type = ["staticlib", "cdylib"]
```

Comment:
concerned about this. does that still work for the release on the different architecture? do we still have the small binary sized libs?

Reply:
Well, after fighting a lot with it. The `-lto` flag only works for `staticlib` and `cdylib`. But if using just those types I wasn't able to make it work as a crate (I mean, importing it from other crates).
The trick is that it only uses those types when compiling. But still allowed me to import the library. Should work well.

Reply:
Maybe this doesn't make sense, but why do we need to go through `wallet-c` and not just directly to `wallet`? Considering that the jni code is in rust, and that we are importing a rust module?

Reply:
Well, the idea is that the exposed functions are already coded in wallet-c. For not duplicating code the `jni` only wraps the calls with the needed types.

Reply:
duplicate the code from the `wallet-c` but don't make it difficult with weird operations.

Reply:
How about I abstract those calls in `wallet` and then use them in `wallet-c`, `wallet-jni` or any other needed place?

Reply:
no because `wallet` exposes the wallet logic in Rust idiomatic way. The C style bindings are mainly wrappers around that to abstract this for the target language idioms (in C it is pointers and functions, in js it is object and classes, in java something else).

now I see what you mean, there is a common components here for all of this targets especially the C and JNI ones. This is why in first place we didn't want to write the jni in rust and reuse the C bindings in first place (now the down side was to write a C++ bindings to bind with the Java...).

What about renaming these end binding directory?

- `/bindings/wallet-core/`: the wallet binding core functions reused for everyone
- `/bindings/wallet-js`: the JS bindings, calling core
- `/bindings/wallet-jni`: the JNI binding, calling core
- `/bindings/wallet-c`: the C bindings
- `/bindings/wallet-android`: the android packaging of the jni bindings in a `AAR` package
- `bindings/...`: ...

Reply:
This way the top level crates are rust driven, and the `bindings/*` are bindings oriented. Might make the root directory a bit simpler too

Reply:
I vote that, that makes it even better yes. Gonna do that. Thanks!

Reply:
Done